python/dgl/sparse/sparse_matrix.py

```diff
@@ -479,6 +479,62 @@ def is_diag(self):
         """Returns whether the sparse matrix is a diagonal matrix."""
         return self.c_sparse_matrix.is_diag()
 
+    def select(self, dim: int, ids: torch.Tensor or slice):
+        """Returns a sub-matrix selected by a list of ids or by range.
+
+        Parameters
+        ----------
+        dim : int
+            The dim to select from matrix, should be 0 or 1.
+        ids : tensor.Tensor or slice
+            Tensor indicates the row ids to select from matrix.
+            NOTE: The ids can include duplicate values.
+            Slice indicates sub-matrix selected by range [start, end).
+
+        The function has not supported autograd.
+
+        Returns
+        -------
+        SparseMatrix
+            The sub-matrix only contains rows or columns selected.
+
+        Examples
+        --------
+        >>> indices = torch.tensor([0, 1, 1, 2, 3, 4], [0, 2, 4, 3, 5, 0]])
+        >>> val = torch.tensor([0, 1, 2, 3, 4, 5])
+        >>> A = dglsp.spmatrix(indices, val)
+        >>> row_ids = torch.tensor([0, 1, 4])
+        >>> A.select(0, row_ids)
+        SparseMatrix(indices=tensor([[0, 1, 1, 2],
+                                     [0, 2, 4, 0]]),
+                     values=tensor([0, 1, 2, 5]),
+                     shape=(3, 6), nnz=4)
+        >>> column_ids = torch.tensor([0, 4, 5])
+        >>> A.select(1, column_ids)
+        SparseMatrix(indices=tensor([[0, 1, 3, 4],
+                                     [0, 1, 2, 0]]),
+                     values=tensor([0, 2, 4, 5]),
+                     shape=(5, 3), nnz=4)
+        >>> A.select(0, slice(1, 3))
+        SparseMatrix(indices=tensor([[0, 0, 1],
+                                     [2, 4, 3]]),
+                     values=tensor([1, 2, 3]),
+                     shape=(2, 6), nnz=3)
+        >>> A.select(1, slice(3, 6))
+        SparseMatrix(indices=tensor([[1, 2, 3],
+                                     [1, 0, 2]]),
+                     values=tensor([2, 3, 4]),
+                     shape=(5, 3), nnz=3)
+        """
+        if isinstance(ids, torch.Tensor):
+            return SparseMatrix(self.c_sparse_matrix.index_select(dim, ids))
+        else:
```

Comment:
if isinstance(ids, slice):
blabla
return

Raise error unsupported input type.

Reply:
I do not find this error type except "TypeError", should I use it?